### ISSUE TITLE
Potential incompatibility with Spectrum's dimensions if not general packet decode failure

### ISSUE THREAD

**Issue by Cynosphere** [Cannoneers-of-Create/CreateBigCannons#450]
Has only occurred randomly but enough to softlock someone in Spectrum's bedrock dimension

Quilt 0.23.1, QFAPI 7.5.0+0.91.0-1.20.1, Spectrum 1.7.4, CBC 0.5.3.b-nightly-8a45d8e
```
Caused by: net.minecraft.class_151: Non [a-z0-9/._-] character in path of location: minecraft:\u0015\u001D\u0003\uFFFD
	at net.minecraft.class_2960.method_45137(class_2960.java:252) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_2960.<init>(class_2960.java:47) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_2960.<init>(class_2960.java:51) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_2960.<init>(class_2960.java:56) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_2540.method_10810(class_2540.java:723) ~[minecraft-1.20.1.i0:0/:?]
	at rbasamoyai.createbigcannons.munitions.big_cannon.fluid_shell.EndFluidStack.readBuf(EndFluidStack.java:48) ~[createbigcannons-0.5.3.b-nightly-8a45d8e.i0:0/:?]
	at rbasamoyai.createbigcannons.index.CBCDataSerializers$1.read(CBCDataSerializers.java:17) ~[createbigcannons-0.5.3.b-nightly-8a45d8e.i0:0/:?]
	at rbasamoyai.createbigcannons.index.CBCDataSerializers$1.method_12716(CBCDataSerializers.java:9) ~[createbigcannons-0.5.3.b-nightly-8a45d8e.i0:0/:?]
	at net.minecraft.class_2945$class_7834.method_46363(class_2945.java:240) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_2945$class_7834.method_46362(class_2945.java:236) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_2739.method_46616(class_2739.java:32) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_2739.<init>(class_2739.java:16) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_2539$class_4532.method_22310(class_2539.java:459) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_2539.method_10783(class_2539.java:522) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_2543.decode(class_2543.java:32) ~[minecraft-1.20.1.i0:0/:?]
	at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:519) ~[netty-codec-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:458) ~[netty-codec-4.1.82.Final.jar:4.1.82.Final]
	... 36 more
```

**Comment by rbasamoyai** [comment]
Something similar was seen with one of the Yungs mods a while back

**Comment by rbasamoyai** [comment]
PR #265 fixed this for Yungs' a while back.
cc @wchen1990 If you're interested in looking into this, feel free to. No idea why this might be happening again.

**Comment by Cynosphere** [comment]
New set of errors get. It would be annoying if its just stacktrace blaming.
<details>
<summary>big big large</summary>
<pre>
[22:09:15] [Render thread/ERROR]: Error executing task on Client
java.lang.IllegalStateException: Invalid entity data item type for field 19 on entity LizardEntity['Lurking Lizard'/59422, l='ClientLevel', x=-692.50, y=-187.00, z=23.50]: old=LizardFrillVariant[texture=spectrum:textures/entity/lizard/frills_simple.png](class de.dafuqs.spectrum.entity.variants.LizardFrillVariant), new=de.dafuqs.spectrum.items.tools.GlassArrowVariant@6b0f1874(class de.dafuqs.spectrum.items.tools.GlassArrowVariant)
	at net.minecraft.class_2945.method_12785(class_2945.java:204) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_2945.method_12779(class_2945.java:191) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_634.method_11093(class_634.java:529) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_2739.method_11808(class_2739.java:46) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_2739.method_11054(class_2739.java:10) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_634.method_48294(class_634.java:2485) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_8042.method_48329(class_8042.java:13) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_8042.method_11054(class_8042.java:6) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_2600.method_11072(class_2600.java:22) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_1255.method_18859(class_1255.java:156) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_4093.method_18859(class_4093.java:23) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_1255.method_16075(class_1255.java:130) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_1255.method_5383(class_1255.java:115) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_310.method_1523(class_310.java:1175) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_310.method_1514(class_310.java:802) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.client.main.Main.main(Main.java:250) ~[minecraft-1.20.1-client.jar:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:568) ~[?:?]
	at org.quiltmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:551) ~[quilt-loader-0.23.1.jar:?]
	at org.quiltmc.loader.impl.launch.knot.Knot.launch(Knot.java:84) ~[quilt-loader-0.23.1.jar:?]
	at org.quiltmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:28) ~[quilt-loader-0.23.1.jar:?]
	at org.prismlauncher.launcher.impl.StandardLauncher.launch(StandardLauncher.java:87) ~[NewLaunch.jar:?]
	at org.prismlauncher.EntryPoint.listen(EntryPoint.java:130) ~[NewLaunch.jar:?]
	at org.prismlauncher.EntryPoint.main(EntryPoint.java:70) ~[NewLaunch.jar:?]
[22:09:15] [Netty Client IO #15/ERROR]: Exception occurred in netty pipeline
io.netty.handler.codec.DecoderException: java.lang.IndexOutOfBoundsException: index: 19, length: 3 (expected: range(0, 20))
	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:489) ~[netty-codec-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:280) ~[netty-codec-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[netty-transport-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[netty-transport-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) ~[netty-transport-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:336) ~[netty-codec-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:308) ~[netty-codec-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[netty-transport-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[netty-transport-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) ~[netty-transport-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:336) ~[netty-codec-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:323) ~[netty-codec-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:444) ~[netty-codec-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:280) ~[netty-codec-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[netty-transport-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[netty-transport-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) ~[netty-transport-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103) ~[netty-codec-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[netty-transport-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[netty-transport-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) ~[netty-transport-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.handler.timeout.IdleStateHandler.channelRead(IdleStateHandler.java:286) ~[netty-handler-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[netty-transport-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[netty-transport-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) ~[netty-transport-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410) ~[netty-transport-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[netty-transport-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[netty-transport-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919) ~[netty-transport-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:166) ~[netty-transport-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:788) ~[netty-transport-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:724) ~[netty-transport-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:650) ~[netty-transport-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562) ~[netty-transport-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997) ~[netty-common-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.1.82.Final.jar:4.1.82.Final]
	at java.lang.Thread.run(Thread.java:833) ~[?:?]
Caused by: java.lang.IndexOutOfBoundsException: index: 19, length: 3 (expected: range(0, 20))
	at io.netty.buffer.AbstractByteBuf.checkRangeBounds(AbstractByteBuf.java:1390) ~[netty-buffer-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.buffer.AbstractByteBuf.checkIndex0(AbstractByteBuf.java:1397) ~[netty-buffer-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.buffer.AbstractByteBuf.checkIndex(AbstractByteBuf.java:1384) ~[netty-buffer-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.buffer.UnsafeByteBufUtil.getBytes(UnsafeByteBufUtil.java:481) ~[netty-buffer-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.buffer.PooledUnsafeDirectByteBuf.getBytes(PooledUnsafeDirectByteBuf.java:130) ~[netty-buffer-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.buffer.ByteBufUtil.decodeString(ByteBufUtil.java:1270) ~[netty-buffer-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.buffer.AbstractByteBuf.toString(AbstractByteBuf.java:1246) ~[netty-buffer-4.1.82.Final.jar:4.1.82.Final]
	at net.minecraft.class_2540.toString(class_2540.java:1728) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_2540.method_10800(class_2540.java:691) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_2540.method_10810(class_2540.java:723) ~[minecraft-1.20.1.i0:0/:?]
	at rbasamoyai.createbigcannons.munitions.big_cannon.fluid_shell.EndFluidStack.readBuf(EndFluidStack.java:48) ~[createbigcannons-0.5.3.b-nightly-8a45d8e.i0:0/:?]
	at rbasamoyai.createbigcannons.index.CBCDataSerializers$1.read(CBCDataSerializers.java:17) ~[createbigcannons-0.5.3.b-nightly-8a45d8e.i0:0/:?]
	at rbasamoyai.createbigcannons.index.CBCDataSerializers$1.method_12716(CBCDataSerializers.java:9) ~[createbigcannons-0.5.3.b-nightly-8a45d8e.i0:0/:?]
	at net.minecraft.class_2945$class_7834.method_46363(class_2945.java:240) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_2945$class_7834.method_46362(class_2945.java:236) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_2739.method_46616(class_2739.java:32) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_2739.<init>(class_2739.java:16) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_2539$class_4532.method_22310(class_2539.java:459) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_2539.method_10783(class_2539.java:522) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_2543.decode(class_2543.java:32) ~[minecraft-1.20.1.i0:0/:?]
	at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:519) ~[netty-codec-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:458) ~[netty-codec-4.1.82.Final.jar:4.1.82.Final]
	... 36 more
</pre>
</details>

**Comment by rbasamoyai** [comment]
Bumping this, but does this issue appear if you remove CBC?

**Comment by Cynosphere** [comment]
No, we've had it removed for a while now and it hasn't come up again.

**Comment by rbasamoyai** [comment]
hmm. Can you replicate this and provide a full crash report?

**Comment by Cynosphere** [comment]
It's never a full crash, just a server disconnect. This is as full of a stacktrace as I can get.
```
[16:02:15] [Netty Client IO #1/ERROR]: Exception occurred in netty pipeline
io.netty.handler.codec.DecoderException: net.minecraft.class_151: Non [a-z0-9/._-] character in path of location: minecraft:\u0015\u001D\u0003\uFFFD
	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:489) ~[netty-codec-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:280) ~[netty-codec-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[netty-transport-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[netty-transport-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) ~[netty-transport-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:336) ~[netty-codec-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:308) ~[netty-codec-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[netty-transport-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[netty-transport-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) ~[netty-transport-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:336) ~[netty-codec-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:323) ~[netty-codec-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:444) ~[netty-codec-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:280) ~[netty-codec-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[netty-transport-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[netty-transport-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) ~[netty-transport-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103) ~[netty-codec-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[netty-transport-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[netty-transport-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) ~[netty-transport-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.handler.timeout.IdleStateHandler.channelRead(IdleStateHandler.java:286) ~[netty-handler-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[netty-transport-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[netty-transport-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) ~[netty-transport-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410) ~[netty-transport-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[netty-transport-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[netty-transport-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919) ~[netty-transport-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:166) ~[netty-transport-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:788) ~[netty-transport-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:724) ~[netty-transport-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:650) ~[netty-transport-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562) ~[netty-transport-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997) ~[netty-common-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.1.82.Final.jar:4.1.82.Final]
	at java.lang.Thread.run(Thread.java:833) ~[?:?]
Caused by: net.minecraft.class_151: Non [a-z0-9/._-] character in path of location: minecraft:\u0015\u001D\u0003\uFFFD
	at net.minecraft.class_2960.method_45137(class_2960.java:252) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_2960.<init>(class_2960.java:47) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_2960.<init>(class_2960.java:51) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_2960.<init>(class_2960.java:56) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_2540.method_10810(class_2540.java:723) ~[minecraft-1.20.1.i0:0/:?]
	at rbasamoyai.createbigcannons.munitions.big_cannon.fluid_shell.EndFluidStack.readBuf(EndFluidStack.java:48) ~[createbigcannons-0.5.3.b-nightly-8a45d8e.i0:0/:?]
	at rbasamoyai.createbigcannons.index.CBCDataSerializers$1.read(CBCDataSerializers.java:17) ~[createbigcannons-0.5.3.b-nightly-8a45d8e.i0:0/:?]
	at rbasamoyai.createbigcannons.index.CBCDataSerializers$1.method_12716(CBCDataSerializers.java:9) ~[createbigcannons-0.5.3.b-nightly-8a45d8e.i0:0/:?]
	at net.minecraft.class_2945$class_7834.method_46363(class_2945.java:240) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_2945$class_7834.method_46362(class_2945.java:236) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_2739.method_46616(class_2739.java:32) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_2739.<init>(class_2739.java:16) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_2539$class_4532.method_22310(class_2539.java:459) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_2539.method_10783(class_2539.java:522) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_2543.decode(class_2543.java:32) ~[minecraft-1.20.1.i0:0/:?]
	at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:519) ~[netty-codec-4.1.82.Final.jar:4.1.82.Final]
	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:458) ~[netty-codec-4.1.82.Final.jar:4.1.82.Final]
	... 36 more
```

**Comment by rbasamoyai** [comment]
Can you provide a full mod list?

**Comment by Cynosphere** [comment]
```
(Fabric)Chiseled-Bookshelf-Visualizer-1.20-2.0.jar
accurate-block-placement-1.2.1.jar
AdaptiveTooltips-1.3.0-fabric-1.20.2.jar
appleskin-fabric-mc1.20.1-2.5.1.jar
architectury-9.1.12-fabric.jar
balm-fabric-1.20-7.0.7.jar
banner-flags-2.1.jar
bellsandwhistles-0.4.5+1.20.1-FABRIC.jar
BetterF3-7.0.2-Fabric-1.20.1.jar
betterlily-1.20-1.3.2-fabric.jar
bettermounthud-1.2.2.jar
blahaj-quilt-1.20.1-0.3.0.jar
Boat-Item-View-Fabric-1.20.1-0.0.5.jar
carryon-fabric-1.20.1-2.1.2.7.jar
cc-tweaked-1.20.1-fabric-1.109.5.jar
cccbridge-mc1.20.1-fabric-v1.6.2b.jar
chalk-2.2.0.jar
chalk-colorful-addon-2.1.1.jar
chat_heads-0.10.31-fabric-1.20.jar
chatcalc-3.0.24b.jar
cloth-config-11.1.118-fabric.jar
Clumps-fabric-1.20.1-12.0.0.3.jar
collective-1.20.1-7.30.jar
Couplings-1.9.4+1.20.jar
create-sodium-fix-0.5.1-d-build.5+mc1.20.1.jar
createbigcannons-fabric-1.20.1-0.5.3.b.jar
createdeco-2.0.1-1.20.1-fabric.jar
Debugify-1.20.1+2.0.jar
disableaccessibilityscreen-1.0.1.jar
ears-fabric-1.20-1.4.6.jar
effective-2.2.0+1.20.1.jar
emi-1.1.0+1.20.1+fabric.jar
emi_enchanting-0.1.0+1.20.1.jar
emi_loot-0.6.5+1.20.1.jar
emiffect-fabric-1.1.2+mc1.20.1.jar
emitrades-fabric-1.2.1+mc1.20.1.jar
emoji-type-2.2.3+1.20.4-fabric.jar
enhancedblockentities-0.9+1.20.jar
entityculling-fabric-1.6.2-mc1.20.1.jar
exposure-1.20.1-1.3.0-fabric.jar
ExtendedDrawers-2.1.1+mc.1.20.1.jar
fabrication-3.4.10+1.20.jar
fabrictailor-2.1.2.jar
fabrishot-1.10.1.jar
ferritecore-6.0.1-fabric.jar
ForgeConfigAPIPort-v8.0.0-1.20.1-Fabric.jar
friendsandfoes-fabric-mc1.20.1-2.0.9.jar
friendsandfoes-flowerymooblooms-fabric-mc1.20.1-2.0.1.jar
handcrafted-fabric-1.20.1-3.0.5.jar
hauntedharvest-1.20-3.1.19-fabric.jar
HorseBuff-1.20-2.1.2.jar
immersive_aircraft-0.7.1+1.20.1-fabric.jar
immersivesnow-1.20.1-1.3.0.jar
indium-1.0.27+mc1.20.1.jar
instantgroup-fabric-1.20.1-1.6.0.jar
InventorySorter-1.9.0-1.20.jar
iris-mc1.20.1-1.6.10.jar
kappa-3.0.4.jar
kleeslabs-fabric-1.20-15.0.0.jar
lambdynamiclights-2.3.2+1.20.1.jar
language-reload-1.5.10+1.20.1.jar
lazydfu-0.1.3.jar
lighty-fabric-2.1.2+1.20.1.jar
lithium-fabric-mc1.20.1-0.11.2.jar
modelfix-1.14-fabric.jar
modmenu-7.2.2.jar
moonlight-1.20-2.9.7-fabric.jar
moyai-1.20-2.0.2-fabric.jar
NoChatReports-FABRIC-1.20.1-v2.2.2.jar
owo-lib-0.11.2+1.20.jar
paginatedadvancements-2.3.0.jar
Patchouli-1.20.1-84-FABRIC.jar
Pehkui-3.7.12+1.14.4-1.20.4.jar
PresenceFootsteps-1.9.1.jar
qfapi-7.5.0_qsl-6.1.2_fapi-0.91.0_mc-1.20.1.jar
quilt-kotlin-libraries-2.1.2+kt.1.9.0+flk.1.10.8.jar
quilt_loading_screen-6.0.0+1.20.1.jar
radio-1.20.1-1.0.0.jar
resourcefullib-fabric-1.20.1-2.1.21.jar
revelationary-1.3.7+1.20.1.jar
scaffoldingdropsnearby-1.20.1-3.2.jar
skinlayers3d-fabric-1.6.2-mc1.20.1.jar
snowysniffer-1.20.1-1.0.0.jar
snowyspirit-1.20-3.0.6-fabric.jar
sodium-fabric-mc1.20.1-0.5.3.jar
sound-physics-remastered-fabric-1.20.1-1.3.1.jar
spectrum-1.7.4-deeper-down.jar
spyglass_astronomy-1.0.8-mc1.20+.jar
Steam_Rails-1.5.1+fabric-mc1.20.1.jar
styled-chat-2.2.3+1.20.1.jar
styled-nicknames-1.2.1+1.20.jar
supplementaries-1.20-2.7.32-fabric.jar
switchy-2.9.1+1.20.jar
switchy-proxy-1.7.0+1.20.jar
switchykit-1.2.0.jar
trinkets-3.7.1.jar
visuality-0.7.1+1.20.jar
vivecraft-1.20.1-1.1.5-fabric.jar
voicechat-fabric-1.20.1-2.5.1.jar
winterly-0.9.3+1.20.jar
Xaeros_Minimap_23.9.7_Fabric_1.20.jar
XaerosWorldMap_1.37.8_Fabric_1.20.jar
yet-another-config-lib-fabric-3.2.1+1.20.jar
yosbr-0.1.2.jar
```

**Comment by rbasamoyai** [comment]
Also, if you have another server lying around somewhere, would like to know if CBC works if Spectrum isn't present. Main thing being, I'm not sure if multiple mods registering EntityDataSerializers may lead to problems

**Comment by rbasamoyai** [comment]
Update: this should have been recently fixed by moving fluid stack syncing to a custom packet as Fabric does not dupport custom data serializer definition. The latest nightly on the discord server should fix this, although it is currently for 1.18.2 only. As there are a few more features for this version on the line, propagation to 1.20.1 will happen later.
